instagram-app/src/App.js

```diff
@@ -0,0 +1,116 @@
+            mydataupdated.liked = !mydataupdated.liked
+            mydataupdated.liked ? mydataupdated.likes = mydataupdated.likes + 1 : mydataupdated.likes = mydataupdated.likes - 1
+          }
+          return mydataupdated;
+        })
+        return {
+          mydata : updatedlike
+        }
+      })
+  }
+
+  //function to trigger user form if there is no local username state
+  postaction = (e, username) => {
+    e.preventDefault();
+    if(this.state.local_user === "") {
+      this.openForm()
+    } else {
+      this.savepost(e, username)
+    }
+  }
+  //function to open the form 
+  openForm () {
+    document.getElementById("myForm").style.display = "block"
+  }
+  //function to close the form
+  closeForm() {
+    document.getElementById("myForm").style.display = "none";
+  }
+  //function to save the state of the user name in the form
+  textchange = (e) => {
+    this.setState({local_user: e.target.value})
+    console.log(e.target.value)
+  }
+  //this is to close the form only.  state is already saved
+  submit = (e) => {
+    e.preventDefault()
+    this.closeForm()
+    console.log(this.state.local_user)
+  }
+  //function to save the state of the post message
+  postchange = (e) => {
+    this.setState({
+      post_comment: e.target.value
+    })
+  }
+  //this is to save the post comment and username
+  savepost = (e, username) => {
+    e.preventDefault()
+    const currentuser = this.state.local_user;
+    const comment = this.state.post_comment;
+    const mydate = Date.now();
+      this.setState(prevState => {
+        const myupdatedpost = prevState.mydata.map(updatepost => {
+          if(updatepost.username === username) {
+            updatepost.comments.push({'com_id':mydate, 'username':currentuser , 'text':comment})
+            console.log(mydate, currentuser , comment)
+          }
+          this.setState({
+            post_comment: ""
+          })
+          return updatepost
+        })
+        return {
+          mydata: myupdatedpost
+        }
+      })
+  }
+
+  render() {
+    return(
+      <div className="app_css">
+      <div className="form-popup" id="myForm">
+        <form onSubmit={(e) =>this.submit(e)} className="form-container">
+          <h2>Enter User Name first</h2>
+          <h3>Click on "Post" after submit</h3>
+          <input onChange={this.textchange} style={{width: '90%'}} className="usernameinput" type="text" placeholder="Enter user name" name="username"></input>
+          <button onClick={this.submit} type="submit" className="btn">Submit</button>
+          <button type="button" className="btn cancel" onClick={this.closeForm}>Close</button>
+        </form>
+      </div>
+
+      <BarSearch />
+        {this.state.mydata.map((postmap, i) => (
+          <PostCont mydata = {postmap} key={postmap.username} postaction={this.postaction} postchange={this.postchange} triggerlikes={this.triggerlikes} post_comment={this.state.post_comment}/>
+        ))}
```

Comment:
All your forms work - great job!

Reply:
The only thing I would change is that I would expect an Enter keystroke to submit my comment...as is, it does not.